xcell/tests/test_data.py

```diff
@@ -48,6 +48,72 @@ def remove_outdir(config):
         os.rmdir(outdir)
 
 
+def get_tracer_pair_iterator(data):
+    tlist = list(data.data['tracers'].keys())
+    for t1 in tlist:
+        for t2 in tlist:
+            yield t1, t2
+
+
+def test_get_tracer_matrix():
+    # No cls from data
+    c = get_config_dict()
+    data = Data(data=c)
+    m = data.get_tracer_matrix()
+    for t1, t2 in get_tracer_pair_iterator(data):
+        assert not m[(t1, t2)]['clcov_from_data']
+    remove_yml_file(c)
+
+    # All cls from data
+    c = get_config_dict()
+    c['cov']['cls_from_data'] = 'all'
+    data = Data(data=c)
+    m = data.get_tracer_matrix()
+    for t1, t2 in get_tracer_pair_iterator(data):
+        assert m[(t1, t2)]['clcov_from_data']
+    remove_yml_file(c)
+
+    # Group cls from data (all)
+    c = get_config_dict()
+    c['cov']['cls_from_data'] = {'DESgc-DESgc': {'compute': 'all'}}
+    data = Data(data=c)
+    m = data.get_tracer_matrix()
+    for t1, t2 in get_tracer_pair_iterator(data):
+        if t1[:5] == 'DESgc' and t2[:5] == 'DESgc':
```

Comment:
You can use `get_tracer_bare_name` https://github.com/xC-ell/xCell/blob/686c4f473d2f6f52bd5f99ebb98903cef7fbf60d/xcell/cls/data.py#L71
or even `get_tracer_bare_name_pair` https://github.com/xC-ell/xCell/blob/686c4f473d2f6f52bd5f99ebb98903cef7fbf60d/xcell/cls/data.py#L76 and do:

```
bname = get_tracer_bare_name_pair(tr1, tr2)
if bname == 'DESgc-DESgc':
```

Reply:
Done!